packages/contracts-ts/scripts/generate-addresses.ts

```diff
@@ -0,0 +1,254 @@
+    proxied: true,
+  },
+} as const
+
+const chains = {
+  1: 'mainnet',
+  10: 'optimism-mainnet',
+  5: 'goerli',
+  420: 'optimism-goerli',
+} as const
+
+if (!glob.sync('node_modules/*').length) {
+  throw new Error(
+    'No node_modules found. Please run `pnpm install` before running this script'
+  )
+}
+
+const nodeModules = path.join(__dirname, '..', 'node_modules')
+
+const bedrockDeployments = {
+  [1]: glob.sync(
+    path.join(nodeModules, '@eth-optimism', 'contracts-bedrock', 'deployments', chains[1], '*.json')
+  ),
+  [10]: glob.sync(
+    path.join(nodeModules, '@eth-optimism', 'contracts-bedrock', 'deployments', chains[10], '*.json')
+  ),
+  [5]: glob.sync(
+    path.join(nodeModules, '@eth-optimism', 'contracts-bedrock', 'deployments', chains[5], '*.json')
+  ),
+  [420]: glob.sync(
+    path.join(nodeModules, '@eth-optimism', 'contracts-bedrock', 'deployments', chains[420], '*.json')
+  ),
+}
+
+// These are here instead of contracts-periphery because contracts-periphery is getting moved to contracts-bedrock eventually so they will be here after that happens
+const peripheryDeployments = {
+  [1]: glob.sync(
+    path.join(nodeModules, '@eth-optimism', 'contracts-periphery', 'deployments', chains[1], '*.json')
+  ),
+  [10]: glob.sync(
+    // contracts-periphery has inconsistent naming and names optimism optimism instead of optimism-mainnet
+    path.join(nodeModules, '@eth-optimism', 'contracts-periphery', 'deployments', 'optimism', '*.json')
+  ),
+  [5]: glob.sync(
+    path.join(nodeModules, '@eth-optimism', 'contracts-periphery', 'deployments', chains[5], '*.json')
+  ),
+  [420]: glob.sync(
+    path.join(nodeModules, '@eth-optimism', 'contracts-periphery', 'deployments', chains[420], '*.json')
+  ),
+}
+
+Object.entries(bedrockDeployments).forEach(([chain, deploymentFiles]) => {
+  if (deploymentFiles.length === 0) {
+    throw new Error(`No bedrock deployments found for ${chains[chain]}`)
+  }
+})
+Object.entries(peripheryDeployments).forEach(([chain, deploymentFiles]) => {
+  if (deploymentFiles.length === 0) {
+    throw new Error(`No periphery deployments found for ${chains[chain]}`)
+  }
+})
+
+const deployments = {
+  [1]: [...bedrockDeployments[1], ...peripheryDeployments[1]],
+  [10]: [...bedrockDeployments[10], ...peripheryDeployments[10]],
+  [5]: [...bedrockDeployments[5], ...peripheryDeployments[5]],
+  [420]: [...bedrockDeployments[420], ...peripheryDeployments[420]],
+}
+
+const getArtifacts = async () => {
+  const artifactPromises: Promise<Deployment & { chainId: number, contractName: string }>[] = []
+  // same loop as 2 for loops 
+  for (const [chainId, deploymentFiles] of Object.entries(deployments)) {
+    for (const artifactPath of deploymentFiles) {
+      const contractName = artifactPath.split('/').reverse()[0]?.replace('.json', '')
+      const artifact = fs.promises.readFile(artifactPath, 'utf8').then((a) => deploymentValidator.parse(JSON.parse(a))).then((deployment) => ({
```

Comment:
why use `.then` instead of `await`?

Reply:
I don't have a personal preference either way